extension/goext/core.go

```diff
@@ -20,6 +20,6 @@ type ICore interface {
 	TriggerEvent(event string, context Context) error
 	HandleEvent(event string, context Context) error
 
-	RegisterEventHandler(eventName string, handler func(context Context, environment IEnvironment) error, priority int)
-	RegisterSchemaEventHandler(schemaID string, eventName string, handler func(context Context, resource Resource, environment IEnvironment) error, priority int)
+	RegisterEventHandler(event string, handler func(context Context, environment IEnvironment) *Error, priority int)
```

Comment:
minor: make handler a first-class type - it will simplify the code when handlers are generated dynamically

Reply:
still not done

Reply:
fixed